SQLQueryGroupProjectTeamB1Julia.sql

```diff
@@ -0,0 +1,255 @@
+@PronounTitle VARCHAR(50),
+@PronounSetName VARCHAR(50),
+@PronounDescription VARCHAR(500)
+
+AS
+
+DECLARE @PRO_ID INT, @PS_ID INT
+
+SET @PS_ID = (SELECT PS.PronounSetID
+			  FROM tblPRONOUN_SET PS
+			  WHERE PS.PSetName = @PronounSetName)
+
+INSERT INTO tblPRONOUN(PronounTitle, PronounsDescr)
+VALUES(@PronounTitle, @PronounDescription)
+
+SET @PRO_ID = SCOPE_IDENTITY()
+
+INSERT INTO tblPRONOUN_PRONOUN_SET(PronounSetID, PronounID)
+VALUES (@PS_ID, @PRO_ID)
+
+--POPULATE TABLES
+
+--Populate the following look-up tables: tblCOMPENSATION, tblQUARTER, tblHOUR
+
+INSERT INTO tblCOMPENSATION(CompName, CompTypeDescr, CompAmount)
+VALUES('Winter or Spring Break', 'Compensation given for taking a shift during Winter or Spring Break', 50),
+		('Holiday Comp', 'Compensation given for taking a shift on a holiday', 75),
+		('Birthday Comp', 'Compensation given for taking a shift on your birthday', 100),
+		('Finals Week Comp', 'Compensation given for taking a shift during finals week', 125),
+		('No Comp', 'No compensation given', 0)
+
+
+INSERT INTO tblQUARTER(QuartName, QuartBegin, QuartEnd, QuartDescr)
+VALUES ('Winter', '01-06-2020', '03-13-2020', 'Winter quarter is the quarter in the beginning of the new year - it is usually the second quarter of the new school year - the weather can be rainy or snowy'),
+		 ('Spring', '03-30-2020', '06-05-2020', 'Spring quarter is usually the third quarter of the new school year - the weather is typically rainy and sunnier towards the end'),
+		 ('Summer', '06-22-2020', '08-21-2020', 'Summer quarter is usually a quarter off for students - it has the warmest weather out of all the quarters'),
+		 ('Summer A', '06-22-2020', '07-22-2020', 'Summer A is the quarter during the first half of summer'),
+		 ('Summer B', '07-23-2020', '08-21-2020', 'Summer B is the quarter during the second half of summer'),
+		 ('Fall', '09-30-2020', '12-11-2020', 'Fall quarter is the quarter in the beginning of the new school year - it is usually the first quarter of the new school year - the weather can be warm, rainy, snowy')
+
+INSERT INTO tblHOUR(HourName)
+VALUES(0), (1), (2), (3), (4), (5), (6), (7), (8), (9), (10), (11), (12), (13), (14), (15), (16), (17), (18), (19), (20), (21), (22), (23)
+GO
+
+--BUSINESS RULES
+
+--Enforce the business rule preventing employees to be scheduled on their birthday
+
+CREATE FUNCTION dbo.BR_NoShiftsOnEmpBirthday()
+RETURNS INT
+AS
+BEGIN
+	DECLARE @RET INT = 0
+	IF EXISTS(SELECT S.ShiftID
+			  FROM tblSHIFT S
+				JOIN tblEMP_SHIFT_STATUS ESS
+					ON S.ShiftID = ESS.ShiftID
+				JOIN tblEMPLOYEE_POSITION EP
+					ON ESS.EmpPosID = EP.EmpPosID
+				JOIN tblEMPLOYEE E
+					ON EP.EmployeeID = E.EmployeeID
+				JOIN tblDAY D
+					ON D.DayID = S.DayID
+				JOIN tblMONTH M
+					ON M.MonthID = S.MonthID
+			  WHERE DAY(E.DOB) = D.[DayName]
+				AND MONTH(E.DOB) = M.MonthNum
+			 )
+	BEGIN
+			SET @RET = 1
+	END
+	RETURN @RET
+END
+GO
+
+ALTER TABLE tblSHIFT
+ADD CONSTRAINT BR_NoShiftsOnBirthday

```

Comment:
For the ‘ALTER TABLE’, should it be tblEMP_SHIFT_STATUS instead of tblSHIFT? Since we want to add constraints to tblEMP_SHIFT_STATUS